homeassistant/components/hydrawise/sensor.py

```diff
@@ -63,15 +120,45 @@ class HydrawiseSensor(HydrawiseEntity, SensorEntity):
 
     def _update_attrs(self) -> None:
         """Update state attributes."""
-        if self.entity_description.key == "watering_time":
-            if (current_run := self.zone.scheduled_runs.current_run) is not None:
-                self._attr_native_value = int(
-                    current_run.remaining_time.total_seconds() / 60
-                )
-            else:
-                self._attr_native_value = 0
-        elif self.entity_description.key == "next_cycle":
-            if (next_run := self.zone.scheduled_runs.next_run) is not None:
-                self._attr_native_value = dt_util.as_utc(next_run.start_time)
-            else:
-                self._attr_native_value = datetime.max.replace(tzinfo=dt_util.UTC)
+        specific_getter = getattr(self, f"_get_{self.entity_description.key}")
+        value = specific_getter()
+        if value is not None:
```

Comment:
`None` should be a valid value, which means you can reduce the method body down to:

`self._attr_native_value = getattr(self, f"_get_{self.entity_description.key}")()`

Reply:
Fixed